src/main/java/hudson/plugins/jira/JiraProjectProperty.java

```diff
@@ -113,25 +114,27 @@ public boolean configure(StaplerRequest req, JSONObject formData) {
             //Should check for existing handler etc, but since this is a dirty hack,
             //we won't
             Stapler.CONVERT_UTILS.deregister(java.net.URL.class);
-            Stapler.CONVERT_UTILS.register(new Converter() {
-                public Object convert(Class aClass, Object o) {
-                    if (o == null || "".equals(o) || "null".equals(o)) {
-                        return null;
-                    }
-                    try {
-                        return new URL((String) o);
-                    } catch (MalformedURLException e) {
-                        LOGGER.warning(String.format("%s is not a valid URL.", o.toString()));
-                        return null;
-                    }
-                }
-            }, java.net.URL.class);
+            Stapler.CONVERT_UTILS.register(new EmptyFriendlyURLConverter(), java.net.URL.class);
             //End hack
 
             sites.replaceBy(req.bindJSONToList(JiraSite.class, formData.get("sites")));
             save();
             return true;
         }
+
+        public static class EmptyFriendlyURLConverter implements Converter {
```

Comment:
NIT: Makes sense to make <code>@Restricted</code>. I doubt it should be exposed to the API, because it still smells like a hack